modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DecimalInput.es.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import propTypes from 'prop-types';
+
+class DecimalInput extends React.Component {
+	static propTypes = {
+		onChange: propTypes.func.isRequired,
+		value: propTypes.oneOfType(
+			[
+				propTypes.string,
+				propTypes.number
+			]
+		)
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			decimal: props.value
+		};
+	}
+
+	_handleDecimalBlur = event => {
+		const value = Number.parseFloat(event.target.value).toFixed(2);
+
+		this.setState(
+			{
+				decimal: value
+			},
+			this.props.onChange(value)
```

Comment:
This might be a brittle to both manage internal state and have an `onChange`. I think it would make more sense to have this be a "controlled" component.

Reply:
@bryceosterhaus  We have an internal state for this input since we're formatting the value onBlur and we don't want to pass an incorrectly formatted value to the parent components. Maybe we could rename the state to `rawValue` or something like that?

Reply:
@thektan Would it make sense to remove internal state and pass both an `onChange` and `onBlur` prop? Just trying to figure out a way to have a "single source of truth" for the value itself

Reply:
I don't think it'd make sense to pass both an onChange and onBlur since that would be creating more conditional logic on a parent component.

Could the value that's being input be similar to a date input where you're changing the month in the dropdown? Where it's an internal state that other components don't need to know about except itself?

Reply:
Hmm, I guess that could make sense. It's just going to be a bit more brittle and could possibly get out of sync with the "real" value, that's my main concern.

Reply:
I see...so ideal scenario: The value is updated in the form's query input on every input change and then changed one last time on blur? I think that makes sense to me. I'll go ahead and make the changes.